cmd/bootstrap/runner.go

```diff
@@ -0,0 +1,185 @@
+package bootstrap
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/giantswarm/apiextensions/v2/pkg/crd"
+	"github.com/giantswarm/backoff"
+	"github.com/giantswarm/k8sclient/v4/pkg/k8sclient"
+	"github.com/giantswarm/k8sclient/v4/pkg/k8srestconfig"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
+	schedulingv1 "k8s.io/api/scheduling/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+)
+
+type runner struct {
+	flag   *flag
+	logger micrologger.Logger
+	stdout io.Writer
+	stderr io.Writer
+}
+
+func (r *runner) Run(cmd *cobra.Command, args []string) error {
+	ctx := context.Background()
+
+	err := r.flag.Validate()
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	err = r.run(ctx, cmd, args)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	return nil
+}
+
+func (r *runner) run(ctx context.Context, cmd *cobra.Command, args []string) error {
+	var err error
+
+	var restConfig *rest.Config
+	{
+		c := k8srestconfig.Config{
+			Logger: r.logger,
+
+			KubeConfig: r.flag.KubeConfig,
+		}
+
+		restConfig, err = k8srestconfig.New(c)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+	}
+
+	k8sClientConfig := k8sclient.ClientsConfig{
+		Logger:     r.logger,
+		RestConfig: restConfig,
+	}
+	k8sClients, err := k8sclient.NewClients(k8sClientConfig)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	err = r.ensureCRDs(ctx, k8sClients)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	err = r.ensurePriorityClass(ctx, k8sClients)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	err = r.ensureNamespace(ctx, k8sClients)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	return nil
+}
+
+func (r *runner) ensureCRDs(ctx context.Context, k8sClients k8sclient.Interface) error {
+	// Ensure Application group CRDs are created.
+	crds := []string{
+		"AppCatalog",
+		"App",
+		"Chart",
+	}
+
+	{
+		for _, crdName := range crds {
+			r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("ensuring %#q CRD exists", crdName))
```

Comment:
```suggestion
			r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("ensuring %#q CRD", crdName))
```
Why not without the `exists` verb?

Reply:
This was also copy pasta I agree reads better without it. Done.